h2o-persist-gcs/src/main/java/water/persist/StringCache.java

```diff
@@ -0,0 +1,33 @@
+package water.persist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+abstract class StringCache {
+  private static final long TIMEOUT_MILLIS = 60 * 1000;
+
+  private long _lastUpdated = 0;
+
+  private List<String> _cache = new ArrayList<>();
+
+  abstract List<String> update();
+
+  List<String> fetch(String resultPrefix, String filter, int limit) {
+    if (System.currentTimeMillis() > _lastUpdated + TIMEOUT_MILLIS) {
+      _cache = update();
+      _lastUpdated = System.currentTimeMillis();
+    }
+
+    int resultCapacity = limit > 0 ? limit : _cache.size();
```

Comment:
This code assumes `_cache` will not be changed during the execution - but `_cache.size()` can actually change in a race-condition (if two threads enter line 17 at the same time).

Reply:
Will be dropped.